### PR TITLE
feat: footer component

### LINKED ISSUE #283: Add footer component
Before building this, we need to evaluate how this will render on pages without a side nav, since all pages need the footer.

image

### PR DESCRIPTION
**Pull request by andreancardona**
Closes #283 

- This PR adds in the footer component 
- Added `date.js` global util

Here is the boxnote with the redlines for reference: https://ibm.box.com/s/d0ppw48bb5v5dycvqrvwvmqkhu0u1hhe

## Testing 

- [ ] Take a look at footer component with Side Nav as can be seen on most pages (look at all breakpoints)
- [ ] Take a look at footer component without the Side Nav as can be seen here: http://localhost:3000/assets/ibmdotcom-react (look at all breakpoints) --> this design should be similar to the [IBMa](https://www.ibm.com/able/) at the different breakpoints.
- Please note we are currently using temporary links

### CODE REVIEW

File: services/web-app/components/footer/footer.js [hunk 1]
```diff
@@ -4,71 +4,104 @@
...
+      text: 'Privacy',
+      link: 'https://github.com/carbon-design-system/carbon-platform'
+    },
+    {
+      text: 'Terms of Use',
+      link: 'https://github.com/carbon-design-system/carbon-platform'
+    },
+    {
+      text: 'IBM.com',
+      link: 'https://github.com/carbon-design-system/carbon-platform'
```
Comment (alisonjoseph) [inline services/web-app/components/footer/footer.js:30]:
```suggestion
      link: 'https://www.ibm.com/'
```
Figured we could add in some of these links?

File: services/web-app/components/footer/footer.js [hunk 2]
```diff
@@ -4,71 +4,104 @@
...
-const Footer = () => {
+const Footer = ({ hasSideNav }) => {
+  const colOne = [
+    {
+      text: 'Contribute',
+      link: 'https://github.com/carbon-design-system/carbon-platform'
+    },
+    {
+      text: 'Privacy',
+      link: 'https://github.com/carbon-design-system/carbon-platform'
```
Comment (alisonjoseph) [inline services/web-app/components/footer/footer.js:22]:
```suggestion
      link: 'https://www.ibm.com/privacy'
```

File: services/web-app/components/footer/footer.js [hunk 3]
```diff
@@ -4,71 +4,104 @@
...
+    },
+    {
+      text: 'Terms of Use',
+      link: 'https://github.com/carbon-design-system/carbon-platform'
```
Comment (alisonjoseph) [inline services/web-app/components/footer/footer.js:26]:
```suggestion
      link: 'https://www.ibm.com/legal'
```

File: services/web-app/components/footer/footer.js [hunk 4]
```diff
@@ -4,71 +4,104 @@
...
+  ]
+
+  const colTwo = [
+    {
+      text: 'Medium',
+      link: 'https://github.com/carbon-design-system/carbon-platform'
+    },
+    {
+      text: 'Twitter',
+      link: 'https://github.com/carbon-design-system/carbon-platform'
```
Comment (alisonjoseph) [inline services/web-app/components/footer/footer.js:41]:
```suggestion
      link: 'https://twitter.com/_carbondesign'
```

File: services/web-app/components/footer/footer.js [hunk 5]
```diff
@@ -4,71 +4,104 @@
...
+                    <a className={clsx(styles.text, styles.link, styles.listLink)}>{item.text}</a>
+                  </Link>
+                </li>
+              ))}
+            </ul>
+          </Column>
+          <Column sm={4} md={3} lg={4} className={styles.column}>
+            <p className={styles.text}>
+              Have questions?{' '}
+              <Link href="https://github.com/carbon-design-system/carbon-platform">
```
Comment (alisonjoseph) [inline services/web-app/components/footer/footer.js:85]:
```suggestion
              <Link href="mailto:carbon@us.ibm.com">
```

File: services/web-app/components/footer/footer.js [hunk 6]
```diff
@@ -4,71 +4,104 @@
...
+                <a className={clsx(styles.text, styles.link)}>Email</a>
+              </Link>{' '}
+              us for site feedback or open an issue in{' '}
+              <Link href="https://github.com/carbon-design-system/carbon-platform">
+                <a className={clsx(styles.text, styles.link)}>Github.</a>
+              </Link>
+            </p>
```
Comment (alisonjoseph) [inline services/web-app/components/footer/footer.js:92]: Do we want to add the copyright here?

`const currentYear = new Date().getFullYear();`

`<p>Copyright © {currentYear} IBM</p>`

File: services/web-app/components/footer/footer.js [hunk 7]
```diff
@@ -4,71 +4,104 @@
...
+            </ul>
+          </Column>
+          <Column sm={4} md={3} lg={4} className={styles.column}>
+            <p className={styles.text}>
+              Have questions?{' '}
+              <Link href="https://github.com/carbon-design-system/carbon-platform">
+                <a className={clsx(styles.text, styles.link)}>Email</a>
+              </Link>{' '}
+              us for site feedback or open an issue in{' '}
+              <Link href="https://github.com/carbon-design-system/carbon-platform">
```
Comment (alisonjoseph) [inline services/web-app/components/footer/footer.js:89]:
```suggestion
              <Link href="https://github.com/carbon-design-system/carbon-platform/issues/new">
```

**Comment by alisonjoseph** [comment]
Added some small content/link suggestions 😄

File: services/web-app/components/footer/footer.js (hunk 6 above)
Comment (andreancardona) [reply services/web-app/components/footer/footer.js:92]: Yep! I spoke with @jeanservaas and this is what will be the content: 

Last updated March 2022 (current month & year)
©2022 IBM Design Program Office

File: services/web-app/components/footer/footer.js [hunk 8]
```diff
@@ -1,74 +1,90 @@
...
+            <ul>{getColList(colOne)}</ul>
+          </Column>
+          <Column sm={4} md={2} lg={4} className={styles.column}>
+            <ul>{getColList(colTwo)}</ul>
+          </Column>
+          <Column sm={4} md={3} lg={3} className={styles.column}>
+            <p className={styles.text}>
+              Last updated {currentMonth} {currentYear}
+              <br />
+              ©2022 IBM Design Program Office
```
Comment (alisonjoseph) [inline services/web-app/components/footer/footer.js:74]:
```suggestion
              ©{currentYear} IBM Design Program Office
```

File: services/web-app/components/footer/footer.js [hunk 9]
```diff
@@ -1,74 +1,90 @@
...
+    <Theme theme="g100" className={styles.container}>
+      <footer>
+        <Grid className={styles.footer}>
+          <Column sm={4} md={2} lg={{ span: 2, start: 5 }} className={styles.column}>
+            <ul>{getColList(colOne)}</ul>
+          </Column>
+          <Column sm={4} md={2} lg={4} className={styles.column}>
+            <ul>{getColList(colTwo)}</ul>
+          </Column>
+          <Column sm={4} md={3} lg={3} className={styles.column}>
```
Comment (alisonjoseph) [inline services/web-app/components/footer/footer.js:70]:
```suggestion
          <Column sm={4} md={3} lg={5} className={styles.column}>
```

I'd change to 5 at large so the copyright text doesn't have a line break at a smaller screen.

File: services/web-app/components/footer/footer.js [hunk 10]
```diff
@@ -1,74 +1,90 @@
...
+            <p className={styles.text}>
+              Last updated {currentMonth} {currentYear}
+              <br />
+              ©2022 IBM Design Program Office
+            </p>
+          </Column>
+          <Column lg={hasSideNav ? { start: 5 } : { start: 1 }} className={styles.logo}>
```
Comment (alisonjoseph) [inline services/web-app/components/footer/footer.js:77]:
```suggestion
          <Column md={{ start: 1 }}  lg={hasSideNav ? { start: 5 } : { start: 1 }} className={styles.logo}>
```
Need to start at 1 at medium or the logo goes off to the right

**Comment by alisonjoseph** [comment]
Couple grid comments, also need @jeanservaas to chime in to confirm correct type token for the footer. Currently looks like you have `body-long-01`.

File: services/web-app/components/footer/footer.js [hunk 11]
```diff
@@ -1,74 +1,90 @@
...
+      </g>{' '}
+    </svg>
+  )
```
Comment (francinelucca) [inline services/web-app/components/footer/footer.js:58]: should this maybe live in its own file? like the thumbnails which we have under `public/assets/thumbnails` , maybe we can put this under `public/assets/ibm-logo.svg`

File: services/web-app/components/footer/footer.js [hunk 12]
```diff
@@ -1,74 +1,90 @@
...
+            <ul>{getColList(colTwo)}</ul>
+          </Column>
+          <Column sm={4} md={3} lg={5} className={styles.column}>
+            <p className={styles.text}>
+              Last updated {currentMonth} {currentYear}
+              <br />
+              ©{currentYear} IBM Design Program Office
+            </p>
+          </Column>
+          <Column md={{ start: 1 }}  lg={hasSideNav ? { start: 5 } : { start: 1 }} className={styles.logo}>
```
Comment (francinelucca) [inline services/web-app/components/footer/footer.js:77]:
```suggestion
          <Column md={{ start: 1 }}  lg={{ start: hasSideNav ? 5 : 1 }} className={styles.logo}>
```

**Comment by andreancardona** [comment]
> ground color doesn't match what we have on

Thanks @alisonjoseph good questions!


here is the sketched redlines for reference also: https://ibm.box.com/s/x72jtusuu91nab1usgmk4kbya1c2x2cm

**Review by francinelucca — CHANGES_REQUESTED** [review]
Minor Comments.
Note: 
- https://www.ibm.com/able/ seems to have more height than this one, not sure if they should be the same height
- Letters seem to be shifting a bit left in `sm` breakpoint, this ok?

**Comment by alisonjoseph** [comment]
The red line for the footer doesn't appear to be using tokens? It just has black as the background color and #f3f3f3 as the link/text color, neither of which map to a token. :sweat_smile: 

@jean.ann.servaas need your help to confirm the footer is using the right color tokens. :slightly_smiling_face:

File: services/web-app/components/footer/footer.js (hunk 11 above)
Comment (andreancardona) [reply services/web-app/components/footer/footer.js]: doone!

**Comment by andreancardona** [comment]
@francinelucca @alisonjoseph 

Made all of the following changes. Please re-review when you get a chance: 

- [ ] updated grid padding in mobile
- [ ] moved IBM logo to `carbon-platform/icons` package
- [ ] updated color tokens: `$background --> $black-100`

##TODO

- [ ] Add IBM logo to box folder mentioned by Matt

**Review by alisonjoseph — APPROVED** [review]
LGTM! 🚀

**Review by francinelucca — APPROVED** [review]
👩‍🍳 💋 !

**Pull request merged**